lifecycle-operator/webhooks/pod_mutator/handlers/appcreationrequest_handler_test.go

```diff
@@ -0,0 +1,244 @@
+			},
+		}}
+
+	singleServiceCreationReq := &klcv1alpha3.KeptnAppCreationRequest{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KeptnAppCreationRequest",
+			APIVersion: "lifecycle.keptn.sh/v1alpha3",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            TestWorkload,
+			Namespace:       namespace,
+			ResourceVersion: "1",
+			Annotations: map[string]string{
+				"keptn.sh/app-type": "single-service",
+			},
+		},
+		Spec: klcv1alpha3.KeptnAppCreationRequestSpec{AppName: TestWorkload},
+	}
+
+	tests := []struct {
+		name    string
+		client  client.Client
+		pod     *corev1.Pod
+		wanterr string
+		wantReq *klcv1alpha3.KeptnAppCreationRequest
+	}{
+		{
+			name:    "Create AppCreationRequest inherit from workload",
+			pod:     pod,
+			client:  fake.NewClient(),
+			wantReq: singleServiceCreationReq,
+		},
+		{
+			name:    "AppCreationRequest already exists",
+			pod:     pod,
+			client:  fake.NewClient(singleServiceCreationReq),
+			wantReq: singleServiceCreationReq,
+		},
+		{
+			name: "Create AppCreationRequest",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "example-pod",
+					Namespace: namespace,
+					Annotations: map[string]string{
+						apicommon.AppAnnotation:      testApp,
+						apicommon.WorkloadAnnotation: TestWorkload,
+						apicommon.VersionAnnotation:  "0.1",
+					},
+				}},
+			client: fake.NewClient(),
+			wantReq: &klcv1alpha3.KeptnAppCreationRequest{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "KeptnAppCreationRequest",
+					APIVersion: "lifecycle.keptn.sh/v1alpha3",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            testApp,
+					Namespace:       namespace,
+					ResourceVersion: "1",
+				},
+				Spec: klcv1alpha3.KeptnAppCreationRequestSpec{AppName: testApp},
+			},
+		},
+		{
+			name: "Error Fetching AppCreationRequest",
+			pod:  &corev1.Pod{},
+			client: k8sfake.NewClientBuilder().WithInterceptorFuncs(interceptor.Funcs{
+				Get: func(ctx context.Context, client client.WithWatch, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+					return errors.New("bad")
+				},
+			}).Build(),
+			wanterr: "could not fetch AppCreationRequest: bad",
+		},
+		{
+			name: "Error Creating AppCreationRequest",
+			pod:  pod,
+			client: k8sfake.NewClientBuilder().WithInterceptorFuncs(interceptor.Funcs{
+				Create: func(ctx context.Context, client client.WithWatch, obj client.Object, opts ...client.CreateOption) error {
+					return errors.New("badcreate")
+				},
+			}).Build(),
+			wanterr: "badcreate",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			appHandler := &AppCreationRequestHandler{
+				Client:      tt.client,
+				Log:         log,
+				EventSender: mockEventSender,
+				Tracer:      tr,
+			}
+			err := appHandler.Handle(context.TODO(), tt.pod, namespace)
+
+			if tt.wanterr != "" {
+				require.NotNil(t, err)
+				require.Contains(t, err.Error(), tt.wanterr)
```

Comment:
if we expect certain error types to be returned, I would prefer to have them checked with `require.ErrorIs(t, err, ErrSomething)`